test/local/mysql_tests.js

```diff
@@ -221,6 +221,105 @@ DB.connect(config)
       )
 
       test(
+        'readMultiple with valid queries',
+        function (t) {
+          db.readMultiple([
+            {
+              sql: 'SELECT * FROM accounts LIMIT ?',
+              params: 1
+            },
+            {
+              sql: 'SELECT COUNT(*) AS count FROM accounts WHERE createdAt < ? AND normalizedEmail LIKE ?',
+              params: [ Date.now(), '%@mozilla.com' ]
+            }
+          ])
+          .then(
+            function(results) {
+              t.plan(9)
+              t.ok(Array.isArray(results), 'results array was returned')
+              t.equal(results.length, 2, 'results array contained two items')
+              t.ok(Array.isArray(results[0]), 'first result was an array')
+              t.ok(results[0].length <= 1, 'first result contained zero or one items')
+              t.ok(Array.isArray(results[1]), 'second result was an array')
+              t.equal(results[1].length, 1, 'second result contained one item')
+              t.equal(typeof results[1][0], 'object', 'second result item was object')
+              t.equal(Object.keys(results[1][0]).length, 1, 'second result item had one property')
+              t.ok(results[1][0].count >= 0, 'count property was non-negative number')
+              t.end()
+            },
+            function(err) {
+              t.plan(1)
+              t.fail(err.stack)
+              t.end()
+            }
+          )
+        }
+      )
+
+      test(
+        'readMultiple with final query',
+        function (t) {
+          t.plan(1)
+          db.readMultiple([
+            { sql: 'SELECT * FROM accounts LIMIT 1' },
+            { sql: 'SELECT * FROM accounts LIMIT 1' }
+          ], { sql: 'SELECT * FROM accounts LIMIT 1' })
+          .then(
+            function(results) {
+              t.equal(results.length, 2, 'results array contained two items')
+              t.end()
+            },
+            function(err) {
+              t.fail(err)
+              t.end()
+            }
```

Comment:
Our `ptaptest` wrapper takes care of this "fail on error, call t.end() when done" logic automatically, just return the promise from the test function.